### ISSUE TITLE
[Deployment Issue] Cannot connect to postgres database in Maven app

### ISSUE THREAD

**Issue by AlfieFitch** [caprover/caprover#2139]
What is the problem?

My maven app cannot connect to my one-click postgres database. I have set up the cloudflare tunnel app and am able to connect to the database through that using the provided 'srv-captain--film-smash-prod-db' address. This makes me think that the database service is available but for some reason my other app just wont connect.

The environment variables I am using in the maven app are: 
POSTGRES_URL=jdbc:postgresql://srv-captain--film-smash-prod-db:5432/postgres

If applicable, content of `captain-definition` file:

{
  "schemaVersion": 2,
  "dockerfileLines": [
      "FROM maven",
      "WORKDIR /workspace/app",
      "COPY pom.xml .",
      "COPY .env .",
      "COPY src src",
      "RUN mvn install -DskipTests",
      "EXPOSE 8080",
      "RUN mvn spring-boot:run"
  ]
}

Output of the following command on your server:

Linux filmsmashvnic 5.15.0-1059-oracle #65~20.04.1-Ubuntu SMP Fri Apr 19 19:01:19 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux

**Comment by githubsaturn** [comment]
Very likely you have a bug in your application. Either the env var is being overridden, or there is a typo etc... 

To debug, you can [shell into your app](https://caprover.com/docs/troubleshooting.html#how-to-run-shell-inside-my-application-inside-container) and run `curl -v srv-captain--film-smash-prod-db` to see if the IP address is being resolved

**Comment by AlfieFitch** [comment]
Thanks for the response -

I ran the command in the app, hitting port 5432 returned:

Connected to srv-captain--film-smash-prod-db (10.0.1.112) port 5432 (#0)
> GET / HTTP/1.1
> Host: srv-captain--film-smash-prod-db:5432
> User-Agent: curl/7.69.1
> Accept: */*
> 
* Empty reply from server

So I assume this means the app can reach the database. I changed the url in the maven app to be the ip address shown there and it still won't connect.

To check that the env variable wasn't be overwritten, I added a print statement before the spring server boots up to output the database url. It shows - jdbc:postgresql://srv-captain--film-smash-prod-db:5432/postgres, which should work?
